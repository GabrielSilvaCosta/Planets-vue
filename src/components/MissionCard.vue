<template>
  <div class="mission-card">
    <p class="mission-year">{{ year }}</p>
    <p class="mission-name">{{ name }}</p>
    <p class="mission-country">{{ country }}</p>
    <p class="mission-destination">{{ destination }}</p>
  </div>
</template>

<script>
export default {
  name: "MissionCard",
  props: {
    year: String,
    name: String,
    country: String,
    destination: String,
  },
};
</script>

<style scoped>
.mission-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
}

.mission-year {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 10px;
}

.mission-name {
  font-size: 18px;
  margin-bottom: 5px;
}

.mission-country {
  font-size: 16px;
  margin-bottom: 5px;
}
</style>
