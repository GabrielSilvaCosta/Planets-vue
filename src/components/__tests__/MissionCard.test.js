import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import MissionCard from "../MissionCard.vue";

describe("MissionCard", () => {
    it("should render correctly", () => {
        const wrapper = mount(MissionCard, {
            props: {
                mission: {
                    name: "Mercurio",
                    year: "Teste1",
                    country: "Teste",
                    destination: "Teste",
                },
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
})