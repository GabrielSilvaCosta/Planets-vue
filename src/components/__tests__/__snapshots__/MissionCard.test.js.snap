// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionCard > should render correctly 1`] = `
"<div data-v-32be5780="" class="mission-card" mission="[object Object]">
  <p data-v-32be5780="" class="mission-year"></p>
  <p data-v-32be5780="" class="mission-name"></p>
  <p data-v-32be5780="" class="mission-country"></p>
  <p data-v-32be5780="" class="mission-destination"></p>
</div>"
`;
